lxd/operations.go

```diff
@@ -251,13 +252,29 @@ func operationDelete(d *Daemon, r *http.Request) response.Response {
 	op, err := operations.OperationGetInternal(id)
 	if err == nil {
 		projectName := op.Project()
-		if op.Permission() != "" {
-			if projectName == "" {
-				projectName = api.ProjectDefaultName
-			}
+		if projectName == "" {
+			projectName = api.ProjectDefaultName
+		}
 
-			if !s.Authorizer.UserHasPermission(r, projectName, op.Permission()) {
-				return response.Forbidden(nil)
+		objectType, entitlement := op.Permission()
+		if objectType != "" {
+			for _, v := range op.Resources() {
+				for _, u := range v {
+					_, _, pathArgs, err := dbCluster.URLToEntityType(u.String())
+					if err != nil {
+						return response.InternalError(fmt.Errorf("Unable to parse operation resource URL: %w", err))
+					}
+
+					object, err := auth.NewObject(objectType, projectName, pathArgs...)
+					if err != nil {
+						return response.InternalError(fmt.Errorf("Unable to create authorization object for operation: %w", err))
+					}
+
+					err = s.Authorizer.CheckPermission(r.Context(), r, object, entitlement)
```

Comment:
Can we use a "get permission checker here" to avoid having to call out to the authorizer service for each op.Resources()?